src/workerd/api/r2-bucket.c++

```diff
@@ -439,7 +460,8 @@ jsg::Promise<kj::Maybe<jsg::Ref<R2Bucket::HeadResult>>> R2Bucket::put(jsg::Lock&
     });
 
     auto& context = IoContext::current();
-    auto client = context.getHttpClient(clientIndex, true, kj::none, "r2_put"_kjc);
+    auto client = r2GetClient(context, clientIndex, "r2_put"_kjc, "PutObject"_kjc,
```

Comment:
I still think that calling the original function with initializer in-place wouldn't increase code amount much (maybe couple lines in total), but would be much more readable:

```
auto client = r2GetClient(context, clientIndex, { { "rpc.method"_kjc, "PutObject"_kjc} ... }
```

because currrently it is very hard to tell what is `r2_put` and what is `PutObject` and what is the difference.